cl/sentinel/service/service.go

```diff
@@ -193,9 +196,11 @@ func (s *SentinelServer) requestPeer(ctx context.Context, pid peer.ID, req *sent
 	if resp.StatusCode < 200 || resp.StatusCode > 399 {
 		errBody, _ := io.ReadAll(resp.Body)
 		errorMessage := fmt.Errorf("SentinelHttp: %s", string(errBody))
-		s.sentinel.Peers().RemovePeer(pid)
-		s.sentinel.Host().Peerstore().RemovePeer(pid)
-		s.sentinel.Host().Network().ClosePeer(pid)
+		if peerCount > gracePeerCount {
```

Comment:
a bit curious `gracePeerCount`, so it's the minimum number we need for working peers? Any reason why we set it 8?